hawk/app/models/report.rb

```diff
@@ -1,69 +1,217 @@
 # Copyright (c) 2009-2015 Tim Serong <tserong@suse.com>
 # See COPYING for license.
 
+require 'fileutils'
+
 class Report
   attr_accessor :id
-  attr_accessor :order
-
+  attr_accessor :name
   attr_accessor :archive
-  attr_accessor :from
-  attr_accessor :until
+  attr_accessor :from_time
+  attr_accessor :to_time
 
   def initialize(attributes)
-    attributes.each do |key, value|
-      self.send(
-        "#{key}=".to_sym,
-        value
-      )
-    end
+    attributes.each { |key, value| send("#{key}=".to_sym, value) }
   end
 
-  def delete
+  def delete(hb_report)
     self.archive.delete
+    FileUtils.remove_entry_secure(hb_report.path) if File.exists?(hb_report.path)
+    FileUtils.remove_entry_secure(hb_report.outfile) if File.exists?(hb_report.outfile)
+    FileUtils.remove_entry_secure(hb_report.errfile) if File.exists?(hb_report.errfile)
   end
 
   def load!
+  end
+
+  def report_path
+    self.class.report_path
+  end
+
+  def mimetype
+    if archive.extname == ".bz2"
+      "application/x-bzip"
+    elsif archive.extname == ".xz"
+      "application/x-xz"
+    elsif archive.extname == ".gz"
+      "application/x-gz"
+    else
+      "application/x-compressed"
+    end
+  end
+
+  def transitions(hb_report)
+    # TODO(fix this)
+    # Have to blow this away if it exists (i.e. is a cached report), else
+    # prior cibadmin calls on individual PE inputs will have wrecked their mtimes.
+    # FileUtils.remove_entry_secure(hb_report.path) if File.exists?(hb_report.path)
 
+    source = archive
+    source = hb_report.path if File.directory?(hb_report.path)
 
+    pcmk_version = nil
+    m = `cibadmin -!`.match(/^Pacemaker ([^ ]+) \(Build: ([^)]+)\)/)
+    pcmk_version = "#{m[1]}-#{m[2]}" if m
 
+    [].tap do |peinputs|
+      peinputs_raw, err, status = Util.capture3("crm", "history", :stdin_data => "source #{source}\npeinputs\n")
```

Comment:
Lint/UselessAssignment: Useless assignment to variable - `err`. Use `_` or `_err` as a variable name to indicate that it won't be used.<br>Style/HashSyntax: Use the new Ruby 1.9 hash syntax.